src/from-car.ts

```diff
@@ -38,6 +38,9 @@ export async function* fromCar (carReader: CarReader|CarIndexedReader, roots?: C
         throw new Error(`Invalid CAR. Hash of block data does not match CID ${cid}`)
       }
       return res
+    },
+    put: ({ cid, bytes }: { cid: CID, bytes: Uint8Array }) => {
```

Comment:
we can land this for the PR, but I'd like to loosen the signature for fromCar back to anyhing CarReaderIsh, as it really doesn't care what kinda thing you give it as long sa it can get blocks by cid and optionally tell you what the roots are. We could make passing in the roots not optional, and then any old blockReader impl would work. If you've not strong objections I'll PR that change after this one lands.

Reply:
Sounds good to me. Just thinking that perhaps we should also PR the new unixfs to have different types for this. One for import and one for export, so that we could use it here?